src/main/java/ca/on/oicr/pineryreports/reports/impl/GeccoReport.java

```diff
@@ -0,0 +1,186 @@
+package ca.on.oicr.pineryreports.reports.impl;
+
+import static ca.on.oicr.pineryreports.util.SampleUtils.*;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.ParseException;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ColumnDefinition;
+import ca.on.oicr.pineryreports.reports.TableReport;
+import ca.on.oicr.ws.dto.RunDto;
+import ca.on.oicr.ws.dto.RunDtoPosition;
+import ca.on.oicr.ws.dto.RunDtoSample;
+import ca.on.oicr.ws.dto.SampleDto;
+import ca.on.oicr.ws.dto.SampleReferenceDto;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.itextpdf.layout.property.TextAlignment;
+
+/**
+ * 
+ * GECCO Report https://jira.oicr.on.ca/browse/GC-875
+ *
+ */
+public class GeccoReport extends TableReport {
+  
+  public static final String REPORT_NAME = "gecco";
+  
+  private final List<ColumnDefinition> columns = Collections.unmodifiableList(Arrays.asList(
+      new ColumnDefinition("Oicr Name", 150f, TextAlignment.LEFT),
+      new ColumnDefinition("Person ID"),
+      new ColumnDefinition("Sample ID"),
+      new ColumnDefinition("Gender"),
+      new ColumnDefinition("Tissue of Origin", 50f, TextAlignment.LEFT),
+      new ColumnDefinition("Tissue Type"),
+      new ColumnDefinition("Libraries Created"),
+      new ColumnDefinition("Times Sequenced"),
+      new ColumnDefinition("Runs")
+  ));
+
+  private Map<String, SampleDto> allSamplesById;
+  private List<SampleDto> geccoTissues;
+  private Map<String, Set<RunDto>> sequencerRunMap;
+  
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    return Collections.emptySet();
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    // No options (could be expanded to report for any project, etc)
+  }
+
+  @Override
+  public String getTitle() {
+    return "GECCO Report";
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
+    List<SampleDto> allSamples = pinery.getSample().all();
+    allSamplesById = mapSamplesById(allSamples);
+    List<RunDto> sequencerRuns = pinery.getSequencerRun().all();
+    sequencerRunMap = sequencerRunHash(sequencerRuns);
+    geccoTissues = filter(filter(allSamples, byProject("GECCO")), bySampleCategory("Tissue"));
+    geccoTissues.sort((dto1, dto2) -> dto1.getName().compareTo(dto2.getName()));
+  }
+
+  private static Map<String, Set<RunDto>> sequencerRunHash(List<RunDto> runs) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)